homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -0,0 +1,51 @@
+"""Coordinator for La Marzocco API."""
+from datetime import timedelta
+import logging
+
+from lmcloud.exceptions import AuthFail, RequestNotSuccessful
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+from .lm_client import LaMarzoccoClient
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LmApiCoordinator(DataUpdateCoordinator[LaMarzoccoClient]):
+    """Class to handle fetching data from the La Marzocco API centrally."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize coordinator."""
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self._lm = LaMarzoccoClient(
+            hass=hass,
+            entry=self.config_entry,
+            callback_websocket_notify=self.async_update_listeners,
+        )
+        self.data = self._lm
+
+    async def _async_update_data(self) -> LaMarzoccoClient:
```

Comment:
You're not really returning any data here. wdyt about just returning None and making the client public here?